src/main/java/org/csanchez/jenkins/plugins/kubernetes/pipeline/ContainerExecDecorator.java

```diff
@@ -268,14 +266,24 @@ public void kill(Map<String, String> modelEnvVars) throws IOException, Interrupt
             private void setupEnvironmentVariable(EnvVars vars, ExecWatch watch) throws IOException
             {
                 for (Map.Entry<String, String> entry : vars.entrySet()) {
-                    watch.getInput().write(
-                            String.format(
-                                    "export %s='%s'%s",
-                                    entry.getKey(),
-                                    entry.getValue().replace("'", "'\\''"),
-                                    NEWLINE
-                            ).getBytes(StandardCharsets.UTF_8)
-                    );
+                    //Check that key is bash compliant.
+                    if (entry.getKey().matches("[a-zA-Z_][a-zA-Z0-9_]*")) {
```

Comment:
I personally would go "uncle bob style" and extract a method with a nice name instead of commenting:

```
if (isLegalShellVariableName(entry.getKey()) {
   ...
```

Comments are very prone to code rot.